http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala

```diff
@@ -90,7 +93,68 @@ abstract class BaseCloudServiceRuntimeMonitor[F[_]] {
                     errorDetails: Option[RuntimeErrorDetails],
                     instances: Set[DataprocInstance])(
     implicit ev: ApplicativeAsk[F, AppContext]
-  ): F[CheckResult]
+  ): F[CheckResult] =
+    for {
+      ctx <- ev.ask
+      _ <- List(
+        // Delete the cluster in Google
+        runtimeAlg
+          .deleteRuntime(
+            DeleteRuntimeParams(
+              runtimeAndRuntimeConfig.runtime.googleProject,
+              runtimeAndRuntimeConfig.runtime.runtimeName,
+              runtimeAndRuntimeConfig.runtime.asyncRuntimeFields.isDefined
+            )
+          )
+          .void, //TODO is this right when deleting or stopping fails?
+        //save cluster error in the DB
+        saveRuntimeError(
+          runtimeAndRuntimeConfig.runtime.id,
+          errorDetails,
+          None //TODO: pass in startup script path in the future
+        ),
+        if (instances.nonEmpty)
+          clusterQuery
+            .mergeInstances(runtimeAndRuntimeConfig.runtime.copy(dataprocInstances = instances))
+            .transaction
+            .void
+        else F.unit
+      ).parSequence_
+
+      // Record metrics in NewRelic
+      _ <- recordStatusTransitionMetrics(
+        monitorContext.start,
+        getRuntimeUI(runtimeAndRuntimeConfig.runtime),
+        runtimeAndRuntimeConfig.runtime.status,
+        RuntimeStatus.Error,
+        runtimeAndRuntimeConfig.runtimeConfig.cloudService
+      )
+
+      // Update the cluster status to Error only if the runtime is non-Deleted.
+      // If the user has explicitly deleted their runtime by this point then
+      // we don't want to move it back to Error status.
+      _ <- for {
+        curStatusOpt <- clusterQuery.getClusterStatus(runtimeAndRuntimeConfig.runtime.id).transaction
+        curStatus <- F.fromOption(
+          curStatusOpt,
+          new Exception(s"Cluster with id ${runtimeAndRuntimeConfig.runtime.id} not found in the database")
+        )
+        _ <- curStatus match {
```

Comment:
This is the main fix: to not change runtime status if it is `Deleted`

Reply:
shall we log a message in this case?

Reply:
👍